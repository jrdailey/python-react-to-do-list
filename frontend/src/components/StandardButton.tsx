const colorMap = {
  blue: 'bg-blue-500',
  gray: 'bg-slate-400',
  green: 'bg-green-500',
  red: 'bg-red-500',
}

type ColorMapKeys = keyof typeof colorMap

interface StandardButtonProps {
  type?: 'button' | 'submit' | 'reset' | undefined,
  color?: ColorMapKeys,
  text: string,
  onClick: () => void,
}

const StandardButton = ({
  type = 'button',
  color = 'gray',
  text,
  onClick,
}: StandardButtonProps) => {
  const bgColor = colorMap[color]

  return (
    <button
      type={type}
      className={`rounded-md px-4 py-2 max-h-[40px] font-semibold text-white opacity-100 cursor-pointer hover:opacity-90 ${bgColor}`}
      onClick={onClick}>
      {text}
    </button >
  )
}

export default StandardButton
