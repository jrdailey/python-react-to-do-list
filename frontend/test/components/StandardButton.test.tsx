import { fireEvent, render, screen } from '@testing-library/react'
import StandardButton from '../../src/components/StandardButton'

describe('StandardButton', () => {
  describe('Default options', () => {
    beforeEach(() => {
      render(<StandardButton text='Button text' />)
    })

    it('matches snapshot', () => {
      const button = screen.getByRole('button')

      expect(button).toMatchSnapshot()
    })

    it('renders the text', () => {
      const button = screen.getByRole('button')

      expect(button).toHaveTextContent('Button text')
    })

    it('renders the button as a `button` type', () => {
      const button = screen.getByRole('button')

      expect(button).toHaveAttribute('type', 'button')
    })

    it('renders the button with a gray color', () => {
      const button = screen.getByRole('button')

      expect(button).toHaveClass('bg-slate-400')
    })
  })

  it('makes use of the given onClick callback', () => {
    const onClick = vi.fn()
    render(<StandardButton text='text' onClick={onClick} />)

    fireEvent.click(screen.getByRole('button'))

    expect(onClick).toHaveBeenCalledOnce()
  })

  it('can render a `submit` button', () => {
    render(<StandardButton type='submit' text='text' />)

    const button = screen.getByRole('button')

    expect(button).toHaveAttribute('type', 'submit')
  })

  it('can render a blue button', () => {
    render(<StandardButton color='blue' text='text' />)

    const button = screen.getByRole('button')

    expect(button).toHaveClass('bg-blue-500')
  })

  it('can render a green button', () => {
    render(<StandardButton color='green' text='text' />)

    const button = screen.getByRole('button')

    expect(button).toHaveClass('bg-green-500')
  })

  it('can render a red button', () => {
    render(<StandardButton color='red' text='text' />)

    const button = screen.getByRole('button')

    expect(button).toHaveClass('bg-red-500')
  })
})
