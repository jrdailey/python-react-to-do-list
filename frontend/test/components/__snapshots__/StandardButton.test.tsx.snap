// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StandardButton > Default options > matches snapshot 1`] = `
<button
  class="rounded-md px-4 py-2 max-h-[40px] font-semibold text-white opacity-100 cursor-pointer hover:opacity-90 bg-slate-400"
  type="button"
>
  Button text
</button>
`;
